### PR TITLE
USE_VEHICLES feature switch

### PR DESCRIPTION
**Pull request by braem**
Closes https://github.com/ChaosInitiative/Chaos-FGD/issues/36

This PR puts vehicle entities, spawnflags, and keyvalues behind `USE_VEHICLES`. `+USE_VEHICLES` to be exact. I also added the `+` for `HL2_ENTITIES`.

The reasoning behind it is that it makes `match_tags` operate more like a feature switch. Without the `+`, an entity/spawnflag/kv is included if it has that tag.

For example, an entity with `appliesto(HL2_ENTITIES, USE_VEHICLES)` is included for P2CE because they have `HL2_ENTITIES`. Using `appliesto(+HL2_ENTITES, +USE_VEHICLES)` makes such an entity not be included for P2CE, because they don't have `USE_VEHICLES`.

### CODE REVIEW

**Comment by braem** [comment]
CI? i can build this fine..

**Comment by braem** [comment]
Workflow fixed in https://github.com/ChaosInitiative/Chaos-FGD/pull/43

File: fgd/bases/Weapon.fgd [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
 
 	// Outputs
 	output OnPlayerUse(void) : "Fires when the player +uses this weapon."
 	output OnPlayerPickup(void) : "Fires when the player picks up this weapon."
 	output OnNPCPickup[-MOMENTUM](void) : "Fires when an NPC picks up this weapon."
-	output OnCacheInteraction[HL2_ENTITIES](void) : "Fires when the player 'proves' they've found this weapon. " +
+	output OnCacheInteraction[+HL2_ENTITIES](void) : "Fires when the player 'proves' they've found this weapon. " +
 		"Fires on: Player Touch, +USE pickup, Physcannon pickup, Physcannon punt."
-	output OnCacheInteraction[-HL2_ENTITIES](void) : "Fires when the player 'proves' they've found this weapon. " +
+	output OnCacheInteraction[-+HL2_ENTITIES](void) : "Fires when the player 'proves' they've found this weapon. " +
```
Comment (Gocnak) [inline fgd/bases/Weapon.fgd:18]: Is this correct here? The -+?

Comment (braem) [reply fgd/bases/Weapon.fgd:18]: Nope! good catch, though im wondering why this sort of thing doesnt throw an error in srctools.

**Review by Gocnak — APPROVED** [review]

**Comment by TeamSpen210** [comment]
Ooh whoops, that's 3.8+ syntax, I need to add the `__future__` import. The tag rule by the way is that all +/- must match, and at least one prefixed name must be present. So you'd normally always use + for features.

**Pull request merged**